src/main/java/org/embulk/input/marketo/rest/MarketoFileResponseEntityReader.java

```diff
@@ -0,0 +1,67 @@
+package org.embulk.input.marketo.rest;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.google.common.io.CharStreams;
+import org.eclipse.jetty.client.api.Response;
+import org.eclipse.jetty.client.util.InputStreamResponseListener;
+import org.eclipse.jetty.http.HttpHeader;
+import org.embulk.input.marketo.exception.MarketoAPIException;
+import org.embulk.input.marketo.model.MarketoResponse;
+import org.embulk.util.retryhelper.jetty92.Jetty92ResponseReader;
+
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Created by tai.khuu on 9/5/17.
+ */
+public class MarketoFileResponseEntityReader implements Jetty92ResponseReader<InputStream>
+{
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+    private InputStreamResponseListener listener;
+
+    private long timeout;
+
+    public MarketoFileResponseEntityReader(long timeout)
+    {
+        this.listener = new InputStreamResponseListener();
+        this.timeout = timeout;
+    }
+
+    @Override
+    public Response.Listener getListener()
+    {
+        return this.listener;
+    }
+
+    @Override
+    public Response getResponse() throws Exception
+    {
+        return this.listener.get(timeout, TimeUnit.MILLISECONDS);
+    }
+
+    @Override
+    public InputStream readResponseContent() throws Exception
+    {
+        if (!getResponse().getHeaders().getField(HttpHeader.CONTENT_TYPE).getValue().equals("text/csv")) {
+            String errorString = readResponseContentInString();
+            MarketoResponse<ObjectNode> errorResponse = OBJECT_MAPPER.readValue(errorString, new TypeReference<MarketoResponse<ObjectNode>>(){ });
```

Comment:
If the `OBJECT_MAPPER` is for reading this `MarketoResponse<>`, I suggest to pre-initialize and keep a `ObjectReader` instead.

Reply:
Fixed